### PR TITLE
Switch from jsonfield to jsonfield2

### PR DESCRIPTION
**Pull request by iamsobanjaved**
Relevant JIRA issue is [here](https://openedx.atlassian.net/browse/BOM-1068).
`edx-celeryutils` is `edx-platofrm`'s dependency, we've to replace `jsonfield` with `jsonfield2` here to ultimately replace it from `edx-platform`.

### CODE REVIEW

**Comment by mraarif** [comment]
@jmbowman the new version of `jsonfield2 3.0.3` drops support for `python2`, we've removed `python2` from here to work with the newer version, is that okay? or there's another plan for removing `python2` support?

File: requirements/base.in [hunk 1]
```diff
@@ -5,6 +5,6 @@
 django-celery
 future
 celery>= 3.1.25,<4.0	
-Django >= 1.11
+Django >= 1.11, < 2.0
```
Comment (mraarif) [inline requirements/base.in]: assuming that we want to continue using `Django 1.11` and use the latest version only for testing

Comment (jmbowman) [reply requirements/base.in]: The problem is that this causes a version conflict when trying to test with Django 2.0+ in edx-platform.  We should handle this like we do in other repos: have `make upgrade` move the testing Django dependency to a separate file, and let tox control the version.

File: requirements/base.in [hunk 2]
```diff
@@ -5,6 +5,6 @@
...
 django-model-utils==3.0.0	
-jsonfield
+jsonfield2==3.0.3
```
Comment (jmbowman) [inline requirements/base.in]: Do we need to pin it?  Our default preference is to let `make upgrade` handle pinning to exact versions whenever possible.

Comment (mraarif) [reply requirements/base.in]: the latest version of `jsonfield2 3.1.0` drops the support for `python 3.5`

File: requirements/base.in (hunk 1 above)
Comment (mraarif) [reply requirements/base.in]: the last `make upgrade` we did, was executed on `python2` and it installed the latest supported version of Django,  now if we run `make upgrade`(with `python3`), it installs `django2.2.x` as the base dependency, my understanding is the base version would still remain `1.11.x`, we can control the version by specifying it in constraints as done in some other repos or it's specified in `edx-platform`. 
As per my understanding, `tox` is testing all Django versions from `1.11.x` to `2.2.x`, could you specify any repo (as you mentioned above "other repos") to follow?

File: requirements/base.in (hunk 2 above)
Comment (jmbowman) [reply requirements/base.in]: Ok, then please move the version constraint to `requirements/constraints.txt` with that explanation in a comment.  Leave the dependency listing here as it was (or maybe add a comment at the end of the line explaining why we depend on it).

File: requirements/base.in (hunk 1 above)
Comment (jmbowman) [reply requirements/base.in]: setup.py [derives](https://github.com/edx/edx-celeryutils/blob/master/setup.py#L81) `install_requires` from this file, so adding the constraint here tells pip in repositories using this library that Django 2.0+ cannot be successfully installed together with it.  edx-platform is one example of the setup I have in mind:

* [make upgrade](https://github.com/edx/edx-platform/blob/master/Makefile#L93-L96)
* [tox.ini](https://github.com/edx/edx-platform/blob/master/tox.ini#L71-L73)

**Pull request merged**